### PR TITLE
[Framework] Delete entities which didn't pass the mapping selector in register_raw

### PR DESCRIPTION
**Pull request by omby8888**
# Description

What - Delete entities which didn't pass the mapping selector in register_raw

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: port_ocean/clients/port/mixins/entities.py [hunk 1]
```diff
@@ -133,9 +134,11 @@ async def batch_delete_entities(
...
         )
 
-    async def search_entities(self, user_agent_type: UserAgentType) -> list[Entity]:
-        query = {
-            "combinator": "and",
+    async def search_entities(
+        self, user_agent_type: UserAgentType, query: dict[Any, Any] | None = None
+    ) -> list[Entity]:
+        default_query = {
+            "combinator": "or",
```
Comment (Tankilevitch) [inline port_ocean/clients/port/mixins/entities.py:141]: seems like the default search entities has changed from `and` to `or` is that on purpose, this has a huge performance impact on the query behavior and might result with different results

File: port_ocean/core/handlers/entity_processor/jq_entity_processor.py [hunk 2]
```diff
@@ -68,15 +74,19 @@ async def _search_as_object(
 
         return result
 
-    async def _get_entity_if_passed_selector(
+    async def _get_mapped_entity(
         self,
         data: dict[str, Any],
         raw_entity_mappings: dict[str, Any],
         selector_query: str,
+        ignore_selector: bool = False,
```
Comment (Tankilevitch) [inline port_ocean/core/handlers/entity_processor/jq_entity_processor.py:82]: there is inconsistency with the parameter across all the functions that pass it through.

File: port_ocean/core/handlers/entity_processor/jq_entity_processor.py [hunk 3]
```diff
@@ -68,15 +74,19 @@ async def _search_as_object(
...
     ) -> dict[str, Any]:
         should_run = await self._search_as_bool(data, selector_query)
-        if should_run:
-            return await self._search_as_object(data, raw_entity_mappings)
+        if ignore_selector or should_run:
+            mapped_entity = await self._search_as_object(data, raw_entity_mappings)
+            mapped_entity[DOES_ENTITY_PASSED_SELECTOR_KEY] = should_run
+            return mapped_entity
```
Comment (Tankilevitch) [inline port_ocean/core/handlers/entity_processor/jq_entity_processor.py:88]: I am not a fan at all of passing flags through unrelated objects, you can change the data structure of how you return that data, we can discuss it offline, but injecting into an object which is user related shouldn't be done, and is very hard to debug.

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
I would prefer that you organize the methods you use, and separate them so its more readable other than trying to re-use existing ones. When using the unregister_raw and register_raw there is no meaning for passing it as diff IIRC

File: port_ocean/clients/port/mixins/entities.py (hunk 1 above)
Comment (omby8888) [reply port_ocean/clients/port/mixins/entities.py:141]: oops...

File: port_ocean/core/handlers/entity_processor/jq_entity_processor.py [hunk 4]
```diff
@@ -124,32 +135,22 @@ async def _calculate_entities(
...
-        entities_after: list[Entity] = await self._calculate_entities(
-            mapping, raw_results["after"]
-        )
-
-        return {
-            "before": entities_before,
-            "after": entities_after,
-        }
+        passed = []
+        failed = []
```
Comment (Tankilevitch) [inline port_ocean/core/handlers/entity_processor/jq_entity_processor.py:146]:
```suggestion
        passed_entities= []
        failed_entities = []
```

File: port_ocean/core/handlers/entity_processor/jq_entity_processor.py [hunk 5]
```diff
@@ -124,32 +135,22 @@ async def _calculate_entities(
                     raw_entity_mappings,
                     mapping.port.items_to_parse,
                     mapping.selector.query,
+                    parse_all,
                 )
             )
-            for data in raw_data
+            for data in raw_results
         ]
         entities = await asyncio.gather(*entities_tasks)
```
Comment (Tankilevitch) [inline port_ocean/core/handlers/entity_processor/jq_entity_processor.py:143]:
```suggestion
        entities_results = await asyncio.gather(*entities_tasks)
```

File: port_ocean/core/handlers/entity_processor/jq_entity_processor.py [hunk 6]
```diff
@@ -124,32 +135,22 @@ async def _calculate_entities(
...
-        )
-
-        return {
-            "before": entities_before,
-            "after": entities_after,
-        }
+        passed = []
+        failed = []
+        for flatten in entities:
+            for entity_data, did_entity_pass_selector in flatten:
```
Comment (Tankilevitch) [inline port_ocean/core/handlers/entity_processor/jq_entity_processor.py:148]:
```suggestion
        for entity_result in entities_results:
            for entity, did_entity_pass_selector in entity_result:
```

Comment (Tankilevitch) [reply port_ocean/core/handlers/entity_processor/jq_entity_processor.py:148]: btw I don't think this unpacking of tuple will work as `flatten` isn't an iterator. 
I think `entity_result[0]` `entity_result[1]` will suit better

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: port_ocean/core/handlers/entity_processor/jq_entity_processor.py (hunk 6 above)
Comment (omby8888) [reply port_ocean/core/handlers/entity_processor/jq_entity_processor.py:148]: i'll make sure but flatten is a list of tuple so it should work

Comment (omby8888) [reply port_ocean/core/handlers/entity_processor/jq_entity_processor.py]: yeah it works

File: port_ocean/core/integrations/mixins/sync_raw.py [hunk 7]
```diff
@@ -135,42 +138,41 @@ async def _register_resource_raw(
...
+                )
+            )
+            if not is_owner:
+                logger.info(
+                    f"Entity {entity_to_delete.identifier} hasn't been created by this integration, "
+                    f"skipping deletion..."
+                )
+                continue
+            await self.entities_state_applier.delete(
+                objects_diff[0]["failed"], user_agent_type
```
Comment (Tankilevitch) [inline port_ocean/core/integrations/mixins/sync_raw.py:162]: this log might be confusing, as it is possible that the entity doesn't exists in port which in that case will return `is_owner=False` as well.

File: port_ocean/core/handlers/entity_processor/jq_entity_processor.py [hunk 8]
```diff
@@ -124,32 +135,22 @@ async def _calculate_entities(
...
-        )
-        entities_after: list[Entity] = await self._calculate_entities(
-            mapping, raw_results["after"]
-        )
-
-        return {
-            "before": entities_before,
-            "after": entities_after,
-        }
+        entities_results = await asyncio.gather(*entities_tasks)
```
Comment (Tankilevitch) [inline port_ocean/core/handlers/entity_processor/jq_entity_processor.py:143]:
```suggestion
        calculate_entities_results = await asyncio.gather(*calculate_entities_tasks)
```

File: port_ocean/core/handlers/entity_processor/jq_entity_processor.py [hunk 9]
```diff
@@ -124,32 +135,22 @@ async def _calculate_entities(
...
+
+        passed_entities = []
+        failed_entities = []
+        for entity_result in entities_results:
+            for entity, did_entity_pass_selector in entity_result:
```
Comment (Tankilevitch) [inline port_ocean/core/handlers/entity_processor/jq_entity_processor.py:148]:
```suggestion
        for entities_results in calculate_entities_results:
            for entity, did_entity_pass_selector in entities_results:
```

File: port_ocean/core/integrations/mixins/sync_raw.py [hunk 10]
```diff
@@ -154,8 +154,8 @@ async def _register_resource_raw(
             )
             if not is_owner:
                 logger.info(
-                    f"Entity {entity_to_delete.identifier} hasn't been created by this integration, "
-                    f"skipping deletion..."
+                    f"Entity {entity_to_delete.identifier} either hasn't been created by this integration "
+                    f"or it doesn't exist, skipping deletion..."
```
Comment (Tankilevitch) [inline port_ocean/core/integrations/mixins/sync_raw.py:158]: Skipping deletion of entity {entity_to_delete.identifier}, couldn't find entity that is related to integration.

**Review by Tankilevitch — APPROVED** [review]
LGTM

File: port_ocean/clients/port/mixins/entities.py [hunk 11]
```diff
@@ -150,6 +153,11 @@ async def search_entities(self, user_agent_type: UserAgentType) -> list[Entity]:
             ],
         }
 
+        if query is None:
+            query = default_query
+        elif query.get("rules"):
+            query["rules"].append(default_query)
+
```
[the remark this answers is not in the document]
Comment (omby8888) [reply port_ocean/clients/port/mixins/entities.py:160]: why?

**Pull request merged**